src/state/opamStateTypes.mli

```diff
@@ -166,6 +166,10 @@ type +'lock switch_state = {
       of removed system dependencies. Only packages which are unavailable end up
       in this set, they are otherwise put in {!field:reinstall}. *)
 
+  overwrote_opams: (bool * OpamFile.OPAM.t) package_map;
```

Comment:
The last commit can be removed if we merge #6531 before. To be squashed otherwise